frontend/src/components/auth/PrivateRoute.tsx

```diff
@@ -11,17 +12,22 @@ type PrivateRouteProps = {
 };
 
 const PrivateRoute: React.FC<PrivateRouteProps> = ({
-  component,
   exact,
   path,
+  component,
 }: PrivateRouteProps) => {
   const { authenticatedUser } = useContext(AuthContext);
 
-  return authenticatedUser ? (
-    <Route path={path} exact={exact} component={component} />
-  ) : (
-    <Redirect to={LOGIN_PAGE} />
-  );
+  if (!authenticatedUser) {
+    return <Redirect to={LOGIN_PAGE} />;
+  }
+  if (
+    (path === TEACHER_PAGE && authenticatedUser.role === "Teacher") ||
+    (path === ADMIN_PAGE && authenticatedUser.role === "Admin")
+  ) {
```

Comment:
I'm a bit confused about what's going on here! Does this just hardcode access for specific roles for specific pages? i.e does someone signed in as "Admin" still have access to the `ComponentLibrary`?